### PR TITLE
Remove inaccurate validation tests.

### LINKED ISSUE #210: Additional validation tests
**[Original report](https://bitbucket.org/glotzer/hoomd-blue/issue/210) by me.**

----------------------------------------

Pull request #278 along with a Jenkins reconfiguration paves the way for long running validation tests as part of hoomd's unit test suite. We need to expand the validation tests with additional tests for MD.

As part of this effort, I would like to simplify the cmake files. Currently we have a LOT of copied and pasted add_test code around. We could simplify that down to a simple macro which all the various cmake scripts call when they want to add a test.

The doumentation needs to be updated to describe the validation testing procedures, including a list of compile options and what hardware / compiler versions we officially validate.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Remove inaccurate validation tests.

## Motivation and context

These tests provide a false sense of correctness. They are not run long enough to sample the chosen state points. We have implemented a new validation test suite, `hoomd-validation` that implements these tests (and more) in a `signac-flow` project that runs on HPC resources.

Resolves #210 

## How has this been tested?

The CI checks pass.

## Change log

Internal change, no change log entry is needed.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
https://github.com/glotzerlab/hoomd-validation is now public and provides a superset of all tests removed in this pull request (some tests are still in open pull requests at the time this message is posted).

File: sphinx-doc/testing.rst [hunk 1]
```diff
@@ -19,12 +19,12 @@ APIs to provide inputs and check for correct outputs. For example, test that the
...
 may take several seconds.
 
-**Validation tests** rigorously check that HOOMD simulations sample the correct
-statistical ensembles. For example, validate the a Lennard-Jones simulation at a
-given density matches the pressure in the NIST reference. Validation tests
-should run long enough to ensure reasonable sampling, but not too long. These
-test run in a CI environment on every pull request. Individual validation tests
-should execute in less than 10 minutes.
+**Long running tests** check for correct behavior, but require up to a minute to execute. Mark long
+running tests with the ``validate`` label.
```
Comment (tommy-waltmann) [inline sphinx-doc/testing.rst:23]: I think it makes sense to keep a label for the longer running tests, but what if we changed the label to be `--long-tests` or something along those lines intead of `--validate`?

**Review by tommy-waltmann — COMMENTED** [review]
I just have one point for discussion about the label for longer tests.

File: sphinx-doc/testing.rst (hunk 1 above)
Comment (joaander) [reply sphinx-doc/testing.rst:23]: This is a breaking change. We could do this in separate pull request for v4. The canonical name would be `slow`.

Comment (tommy-waltmann) [reply sphinx-doc/testing.rst:23]: Sounds good.

**Review by tommy-waltmann — APPROVED** [review]

**Review by Charlottez112 — APPROVED** [review]

**Pull request merged**